### ISSUE TITLE
Need explanation regarding the loss

### ISSUE THREAD

**Issue by SravanChittupalli** [OreoChocolate/LAIN#7]
First off, amazing work — and thank you for open-sourcing the code!

I had a question about the way the loss is computed. I understand much of the code is adapted from Zhang’s UPT repository, but I was hoping you could share some insights into the loss function design choice here.

As you know, the HICO-DET ground truth doesn’t provide exhaustive human–object interaction triplets for every pair in an image. For the below image only annotation between human and elephant are given. The loss calculations for these pairs make sense. However, from what I observed, the loss seems to push the logits for non-annotated human–object pairs (e.g., human–backpack in the example I attached) toward zero, even when they might not actually represent negatives. Wouldn’t this penalize the model for unannotated but potentially valid interactions?

Is this suppression intentional for regularization or negative mining purposes, or am I misunderstanding the implementation?

Thank you so much for your time and for making this code available — it’s been super helpful!

<img width="225" height="224" alt="Image" src="https://github.com/user-attachments/assets/2facc8d8-ce1b-4ec6-9eab-b8caa41e1999" />
